### DISCUSSION TITLE
authorizationURL Returns Private Provider URL

### DISCUSSION THREAD

**Discussion by corinz** [panva/openid-client#521] [Q&A] [answered]
I create my node openid client "client" with an internal provider URL, "provider". I then attempt to auth in my front-end app to my provider through a redirect link that the client provides. When my app is redirected to the my provider, the internal url is returned and the app cannot resolve it. How can I design around this?

So, my app GETs a /login route from my client
My client returns a redirect to the provider authorizationUrl
The authorizationUrl is the internal Url

For context, I am running my client in kube, with my oidc provider. My client would not be able to resolve the internet Url for my provider that my app is expecting. So, I need to config it with private, but return a public. Any ideas?

@panva Any help, is much appreciated. Thank you.

**Comment by panva** [comment, marked as answer]
@corinz configure your Issuer instance authorization_endpoint to be the "internet" one. You will probably not be able to use discovery but you can still construct the Issuer manually with just the metadata.

**Reply by corinz** [reply]
Thank you!
